src/pages/storage/CustomVolumeSelectModal.tsx

```diff
@@ -30,6 +30,7 @@ const CustomVolumeSelectModal: FC<Props> = ({
     data: volumes = [],
     error,
     isLoading,
+    isFetching,
```

Comment:
That is an interesting point, I didn't test this before. Will investigate why the cache is not directly used.

Reply:
There is a prop `refetchOnMount` for reach query, which defaults to `true`. I will set it to `false` here to avoid the refresh, as it will be very visible otherwise.

Reply:
great, now I know this too :smiley: that prop would be definitely relevant in some places

Reply:
Had to set `refetchOnMount: (query) => query.state.isInvalidated,`, so it refetches on device creation (where we explicitly invalidate the query) but reuses the cache on cancel or reopen of the modal.